src/middlewares/errorMiddleware.js

```diff
@@ -0,0 +1,16 @@
+import { ApiError } from '../exception/api.error.js';
+
+export const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
```

Comment:
There is a missing 'else' statement here. If the error is an instance of ApiError, the response is sent, and the function should return to prevent sending another response. Consider adding an 'else' before setting the statusCode to 500.